activemodel/lib/active_model/secure_password.rb

```diff
@@ -61,23 +68,60 @@ def has_secure_password(options = {})
           raise
         end
 
+        attr_reader attribute
+
+        # Encrypts the password into the +password_digest+ attribute, only if the
+        # new password is not empty.
+        #
+        #   class User < ActiveRecord::Base
+        #     has_secure_password validations: false
+        #   end
+        #
+        #   user = User.new
+        #   user.password = nil
+        #   user.password_digest # => nil
+        #   user.password = 'mUc3m00RsqyRe'
+        #   user.password_digest # => "$2a$10$4LEA7r4YmNHtvlAvHhsYAeZmk/xeUVtMTYqwIvYY76EW5GUqDiP4."
+        define_method("#{attribute}=") do |unencrypted_password|
+          if unencrypted_password.nil?
+            self.send("#{attribute}_digest=", nil)
+          elsif !unencrypted_password.empty?
+            instance_variable_set("@#{attribute}", unencrypted_password)
+            cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST : BCrypt::Engine.cost
+            self.send("#{attribute}_digest=", BCrypt::Password.create(unencrypted_password, cost: cost))
+          end
+        end
+
+        define_method("#{attribute}_confirmation=") do |unencrypted_password|
+          instance_variable_set("@#{attribute}_confirmation", unencrypted_password)
+        end
+
+        define_method("regenerate_#{attribute}") do
+          self.send("#{attribute}=", self.class.generate_unique_secure_token)
+        end
```

Comment:
We [still](https://www.reddit.com/r/rails/comments/55f60p/code_reviewfeedback_a_more_secure_version_of_has/d8aa26e) don't need this.